### PR TITLE
SW-3893 Add tabs to tracking v2 withdrawals page

### PR DESCRIPTION
**Pull request by karthikbtf**
- refactor existing withdrawals page to extract out the table content into NurseryWithdrawalsTable
- create tabs in v2 plantings and withdrawals
- reuse NurseryWithdrawalsTable in withdrawal history tab
- instantiate empty plantings component

Terraware App 2023-07-17 07-09-00

Terraware App 2023-07-17 07-08-33

### CODE REVIEW

File: src/components/NurseryWithdrawals/NurseryPlantingsAndWithdrawals.tsx [hunk 1]
```diff
@@ -1,15 +1,53 @@
...
+                </Typography>
+              </Grid>
+            </Grid>
+          </PageHeaderWrapper>
+          <Grid item xs={12}>
+            <PageSnackbar />
+          </Grid>
+          <Box ref={contentRef} display='flex' flexDirection='column' flexGrow={1}>
+            <Card flushMobile style={{ minWidth: 'fit-content' }}>
+              <Tabs
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/NurseryPlantingsAndWithdrawals.tsx:41]: tab content

File: src/components/NurseryWithdrawals/NurseryPlantingsAndWithdrawals.tsx [hunk 2]
```diff
@@ -1,15 +1,53 @@
...
+import PlantingProgress from './PlantingProgressTabContent';
+import NurseryWithdrawals from './NurseryWithdrawalsTabContent';
 
 export default function NurseryPlantingsAndWithdrawals(): JSX.Element {
+  const theme = useTheme();
+  const contentRef = useRef(null);
+
   /**
-   * TODO: Add tabs, etc.
+   * TODO: initialize data, url params etc.
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/NurseryPlantingsAndWithdrawals.tsx:20]: we would dispatch requests here i think

File: src/components/NurseryWithdrawals/NurseryWithdrawals.tsx [hunk 3]
```diff
@@ -1,215 +1,15 @@
...
-import FilterGroup, { FilterField } from 'src/components/common/FilterGroup';
-
-const useStyles = makeStyles((theme: Theme) => ({
-  searchField: {
-    width: '300px',
-  },
-  popoverContainer: {
-    '& .MuiPaper-root': {
-      border: `1px solid ${theme.palette.TwClrBaseGray300}`,
-      borderRadius: '8px',
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/NurseryWithdrawals.tsx:30]: all of this was extracted out into NurseryWithdrawalsTable

File: src/components/NurseryWithdrawals/NurseryWithdrawalsTabContent.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { Typography, useTheme } from '@mui/material';
+import strings from 'src/strings';
+import Card from 'src/components/common/Card';
+import NurseryWithdrawalsTable from './NurseryWithdrawalsTable';
+
+export default function PlantingProgress(): JSX.Element {
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/NurseryWithdrawalsTabContent.tsx]: withdrawals table, some style of title and missing description etc we can fix as follow up, not scope of this PR

File: src/components/NurseryWithdrawals/NurseryWithdrawalsTable.tsx [hunk 5]
```diff
@@ -0,0 +1,280 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/NurseryWithdrawalsTable.tsx:1]: this was extracted out from NurseryWithdrawals. no new code added

File: src/components/NurseryWithdrawals/PlantingProgressTabContent.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { Typography, useTheme } from '@mui/material';
+import strings from 'src/strings';
+import Card from 'src/components/common/Card';
+
+export default function PlantingProgress(): JSX.Element {
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/PlantingProgressTabContent.tsx:5]: empty planting progress tab content

File: package.json [hunk 7]
```diff
@@ -14,7 +14,7 @@
     "@mui/styled-engine-sc": "^5.12.0",
     "@mui/styles": "^5.11.9",
     "@reduxjs/toolkit": "^1.9.3",
-    "@terraware/web-components": "^2.3.48",
+    "@terraware/web-components": "^2.3.51",
```
Comment (karthikbtf) [inline package.json:17]: for tabs

File: src/components/NurseryWithdrawals/NurseryWithdrawalsTabContent.tsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+      <Typography
+        sx={{
+          fontSize: '20px',
+          fontWeight: 600,
+          color: theme.palette.TwClrTxt,
+        }}
+      >
+        {strings.WITHDRAWAL_HISTORY}
+      </Typography>
+      <NurseryWithdrawalsTable />
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/NurseryWithdrawalsTabContent.tsx:20]: reuse nursery withdrawals table

File: src/components/NurseryWithdrawals/NurseryWithdrawalsTabContent.tsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import { Typography, useTheme } from '@mui/material';
+import strings from 'src/strings';
+import Card from 'src/components/common/Card';
+import NurseryWithdrawalsTable from './NurseryWithdrawalsTable';
+
+export default function NurseryWithdrawalsTabContent(): JSX.Element {
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/NurseryWithdrawalsTabContent.tsx:6]: styling of title, missing description etc are out of scope of this PR, we can followup later

File: src/components/NurseryWithdrawals/NurseryPlantingsAndWithdrawals.tsx [hunk 10]
```diff
@@ -1,15 +1,53 @@
...
-      <NurseryWithdrawals />;
-    </>
+    <TfMain>
+      <Box sx={{ paddingLeft: theme.spacing(3) }}>
+        <Grid container spacing={3} sx={{ marginTop: 0 }}>
+          <PageHeaderWrapper nextElement={contentRef.current}>
+            <Grid container spacing={3} sx={{ paddingLeft: theme.spacing(3), paddingBottom: theme.spacing(4) }}>
+              <Grid item xs={8}>
+                <Typography sx={{ marginTop: 0, marginBottom: 0, fontSize: '24px', fontWeight: 600 }}>
+                  {strings.WITHDRAWAL_LOG}
```
Comment (constanzauanini) [inline src/components/NurseryWithdrawals/NurseryPlantingsAndWithdrawals.tsx:31]: This title should be withdrawals

**Review by constanzauanini — APPROVED** [review]
One comment but looking perfect. Thanks for doing this first important step!

**Pull request merged**